### PR TITLE
Add IDPF test vector

### PR DESCRIPTION
**Pull request by divergentdave**
This is a refresh of #132, rebased on top of #160. I changed it to conditionally override one of the IDPF keys to be `b"\x02" * length`, so that we get a more interesting public share, without runs of zeros. I also checked in the resulting test vector. This has knock-on effects on the Poplar1 test vector, naturally.

### CODE REVIEW

File: poc/idpf_poplar.sage [hunk 1]
```diff
@@ -34,6 +35,8 @@ class IdpfPoplar(Idpf):
             gen_rand(IdpfPoplar.Prg.SEED_SIZE),
             gen_rand(IdpfPoplar.Prg.SEED_SIZE),
         ]
+        if TEST_VECTOR:
```
Comment (cjpatton) [inline poc/idpf_poplar.sage:38]: To remind us to delete these lines before copy-pasting into the spec.
```suggestion
        # REMOVE ME
        if TEST_VECTOR:
```

**Review by cjpatton — CHANGES_REQUESTED** [review]
This seems mostly reasonable. My only concern is that the exception in the test vector generation code contradicts this statement in the spec:

> To make the tests deterministic, `gen_rand()` was replaced with a function that returns the requested number of octets.

We need to figure out a way to spell out how the random values are picked.

It may not be strictly necessary to fix this for 04, but it would certainly be nice to have by 05.

**Comment by divergentdave** [comment]
Hmm, I suppose if we want to specify which random bytes go where, then making it an explicit input is the simplest way to do so. Sounds good to me.

**Comment by cjpatton** [comment]
Oops, actually we will want to re-open this once I've addressed #173.

**Pull request closed without merging**